code/datums/gamemode/role/rambler.dm

```diff
@@ -0,0 +1,91 @@
+/datum/role/rambler
+	name = RAMBLER
+	id = RAMBLER
+	required_pref = ROLE_MINOR
+	special_role = RAMBLER
+	logo_state = "rambler-logo"
+	wikiroute = ROLE_MINOR
+
+/datum/role/rambler/OnPostSetup()
+	. =..()
+	if(ishuman(antag.current))
+		equip_rambler(antag.current)
+		name_rambler(antag.current)
+
+/datum/role/rambler/ForgeObjectives()
+	AppendObjective(/datum/objective/investigate)
+	AppendObjective(/datum/objective/freeform/ponder)
+	AppendObjective(/datum/objective/ramble)
+
+/datum/role/rambler/Greet(var/greeting,var/custom)
+	if(!greeting)
+		return
+
+	var/icon/logo = icon('icons/logos.dmi', logo_state)
+	switch(greeting)
+		if (GREET_CUSTOM)
+			to_chat(antag.current, "<img src='data:image/png;base64,[icon2base64(logo)]' style='position: relative; top: 10;'/> <span class='danger'>[custom]</span>")
+		else
+			to_chat(antag.current, "<img src='data:image/png;base64,[icon2base64(logo)]' style='position: relative; top: 10;'/> <span class='danger'>You are a Soul Rambler.<br>You wander space, looking for the man who killed your closest friend. And, perhaps, seeking answers to less tangible questions about life. If you happen to help people along the way, so be it.</span>")
+
+	to_chat(antag.current, "<span class='danger'>You have no powers, except the power to blow minds. Your shakashuri can be used to pacify spirits, but you will otherwise need to rely on your weapons-grade philosophical insights.</span>")
+
+/obj/item/device/instrument/recorder/shakashuri
+	name = "Shakashuri"
+	desc = "Similar to other woodwinds, though it can be played only by a true rambler of souls."
+	exudes_peace = TRUE
+	slot_flags = SLOT_BACK
+
+/obj/item/device/instrument/recorder/shakashuri/attack_self(mob/user)
+	if(!isrambler(user))
+		to_chat(user,"<span class='warning'>Playing this would require 9 years of training!</span>")
+		return
+	else
+		..()
+
+/obj/item/weapon/reagent_containers/food/snacks/quiche/frittata/New()
+	..()
+	name = "frittata"
+	desc = "Now you have no need to eat balls for breakfast."
+	reagents.add_reagent(DOCTORSDELIGHT,37)
+
+/obj/item/clothing/mask/necklace/crystal
+	name = "crystal necklace"
+	desc = "Your final momento of a lost friend. Use it to decide on the killer."
+	icon_state = "tooth-necklace"
+	var/mob/living/suspect = null
```

Comment:
This may cause problems with the guy's deletion.
Probably better to store a ckey as an indirect reference.

Reply:
Could possibly target his mind, since minds don't get deleted afaik?

Reply:
Minds get deleted much rarely, but I think it's better in this case to store a ckey and then compare that ckey with the actual ckey of the traitor.

Reply:
It should definitely be the mind, in case the traitor dies and becomes a diona, xeno, etc.